packages/perseus/src/widgets/interactive-graphs/mafs-graph.tsx

```diff
@@ -149,6 +151,19 @@ export const StatefulMafsGraph = React.forwardRef<
         );
     }, [dispatch, xMinRange, xMaxRange, yMinRange, yMaxRange]);
 
+    const numSegments = graph.type === "segment" ? graph.numSegments : null;
+    const originalPropsRef = useRef(props);
+    const latestPropsRef = useLatestRef(props);
+    useEffect(() => {
+        // This conditional prevents the state from being "reinitialized" right
+        // after the first render. This is an optimization, but also prevents
+        // a bug where the graph would be marked "incorrect" during grading
+        // even if the user never interacted with it.
```

Comment:
I feel like there's probably a simpler way to achieve this that doesn't require keeping a ref to two `props` objects. Maybe I should extract a `useEffectExceptOnMount` hook — unless we already have one?

Reply:
Looking at the [exports from wonder-blocks-core](https://github.com/Khan/wonder-blocks/blob/main/packages/wonder-blocks-core/src/index.ts), I'm not sure there's something that fits your use case. Maybe `useIsMounted` is the closest one? But I don't think that's the same thing... 🤔

Reply:
Question: What's the downside to having the two refs? If you can extract a hook that makes things simpler, that always sounds better to me :) Unless it is harder to follow of course.